internal/etag/reader.go

```diff
@@ -153,3 +167,41 @@ type VerifyError struct {
 func (v VerifyError) Error() string {
 	return fmt.Sprintf("etag: expected ETag %q does not match computed ETag %q", v.Expected, v.Computed)
 }
+
+// UUIDHash - use uuid to make md5sum
+type UUIDHash struct {
```

Comment:
```suggestion
type UUIDHash struct {
         hash.Hash
```

and overriding relevant methods is enough? @aead